storageapi/src/tests/mbusprot/storageprotocoltest.cpp

```diff
@@ -32,183 +36,108 @@ using document::test::makeBucketSpace;
 using storage::lib::ClusterState;
 using vespalib::string;
 
-namespace storage {
-namespace api {
+namespace vespalib {
+
+// Needed for GTest to properly understand how to print Version values.
+// If not present, it will print the byte values of the presumed memory area
+// (which will be overrun for some reason, causing Valgrind to scream at us).
+void PrintTo(const vespalib::Version& v, std::ostream* os) {
+    *os << v.toString();
+}
+
+}
+
+namespace storage::api {
 
-struct StorageProtocolTest : public CppUnit::TestFixture {
+struct StorageProtocolTest : TestWithParam<vespalib::Version> {
     document::TestDocMan _docMan;
     document::Document::SP _testDoc;
     document::DocumentId _testDocId;
+    document::BucketId _bucket_id;
     document::Bucket  _bucket;
+    document::BucketId _dummy_remap_bucket{17, 12345};
+    BucketInfo _dummy_bucket_info{1,2,3,4,5, true, false, 48};
     vespalib::Version _version5_0{5, 0, 12};
     vespalib::Version _version5_1{5, 1, 0};
     vespalib::Version _version5_2{5, 93, 30};
     vespalib::Version _version6_0{6, 240, 0};
+    vespalib::Version _version7_0{7, 0, 0}; // FIXME
```

Comment:
All these version variables can be removed?

Reply:
Done